src/hooks/useUniversalRouter.ts

```diff
@@ -33,55 +35,72 @@ export function useUniversalRouterSwapCallback(
   const { account, chainId, provider } = useWeb3React()
 
   return useCallback(async (): Promise<TransactionResponse> => {
-    try {
-      if (!account) throw new Error('missing account')
-      if (!chainId) throw new Error('missing chainId')
-      if (!provider) throw new Error('missing provider')
-      if (!trade) throw new Error('missing trade')
+    return trace(
+      'swap.send',
+      async ({ setTraceData, setTraceStatus, setTraceError }) => {
+        try {
+          if (!account) throw new Error('missing account')
+          if (!chainId) throw new Error('missing chainId')
+          if (!provider) throw new Error('missing provider')
+          if (!trade) throw new Error('missing trade')
 
-      const { calldata: data, value } = SwapRouter.swapERC20CallParameters(trade, {
-        slippageTolerance: options.slippageTolerance,
-        deadlineOrPreviousBlockhash: options.deadline?.toString(),
-        inputTokenPermit: options.permit,
-        fee: options.feeOptions,
-      })
-      const tx = {
-        from: account,
-        to: UNIVERSAL_ROUTER_ADDRESS(chainId),
-        data,
-        // TODO: universal-router-sdk returns a non-hexlified value.
-        ...(value && !isZero(value) ? { value: toHex(value) } : {}),
-      }
+          setTraceData('slippageTolerance', options.slippageTolerance.toFixed(2))
+          const { calldata: data, value } = SwapRouter.swapERC20CallParameters(trade, {
+            slippageTolerance: options.slippageTolerance,
+            deadlineOrPreviousBlockhash: options.deadline?.toString(),
+            inputTokenPermit: options.permit,
+            fee: options.feeOptions,
+          })
+          const tx = {
+            from: account,
+            to: UNIVERSAL_ROUTER_ADDRESS(chainId),
+            data,
+            // TODO: universal-router-sdk returns a non-hexlified value.
```

Comment:
what exactly is the TODO part of this comment?😅 like make universal-router-sdk return a hexlified value or something else?

Reply:
oh this isn't part of this PR, it's just moved around

Reply:
this is why TODOs should have usernames attached 😆

Reply:
yes, I'd call this a bug in universal-router-sdk